docs/tenancy/1.x/hooks-custom.md

```diff
@@ -0,0 +1,242 @@
+{
+    public $fires = false;
+
+    public function for($event)
+    {
+        parent::for($event);
+        
+        if($this->event instanceof \Tenancy\Tenant\Events\Created)
+        {
+            $this->fires = true;
+        }
+        return $this;
+    }
+}
+```
+
+For more advance or tenant dependent changes to the event consider creating an [Event and Listener](#configureshooks-event) to configure the Hook.
+
+### Differences
+
+The implementation of both `Hook` and `ConfigurableHook` is generally the same with the exception of the following differences.
+
+**Queues**
+
+`Hook` defaults to running on a queue, and defaults to the applications default queue.
+`ConfigurableHook` does not default to using a queue and will only run on a queue when the queue is specified
+
+```php
+// Hook
+public $queued = true;
+public $queue = null;
+
+// ConfigurableHook
+public $queue = null;
+```
+
+**Fires**
+
+`Hook` defaults to always fire as long as the event being passed is a valid Tenancy Event.
+`ConfigurableHook` defaults to always fire.
+
+```php
+// Hook
+public $event; //Must be an instance of Tenancy\Tenant\Events\Event
+public $fires = true;
+
+// ConfigurableHook
+public $fires = true;
+```
+
+> Note: Both `Hook` and `ConfigurableHook` have the `$event` property which contains the tenant in `$event->tenant`
+
+## Registration
+
+There are two methods of registering your custom Lifecycle Event.
+
+### Extend HooksProvider
+
+The simplest method is to create a new Service Provider (or use an existing one) and have it extend `Tenancy\Support\HooksProvider`. Then create an array of your custom hooks.
+
+```php
+<?php
+
+namespace App\Providers;
+
+class TenantLifecycleProvider extends \Tenancy\Support\HooksProvider
+{
+    protected $hooks = [
+        \App\Hooks\CustomHook::class,
+    ];
+}
+```
+
+> This is the recommended method of registering Lifecycle Events
+
+### Resolving the Resolver
+
+```php
+<?php
+
+namespace App\Providers;
+
+use Illuminate\Support\ServiceProvider;
+use Tenancy\Lifecycle\Contracts\ResolvesHooks;
+
+class AppServiceProvider extends ServiceProvider
+{
+    public function boot()
+    {
+        $this->app->resolving(ResolvesHooks::class, function (ResolvesHooks $resolver) {
+            $resolver->addHook(\App\Hooks\CustomHook::class);
+        });
+    }
+}
+```
+
+> Please note, this is not the recommended method of registering your Lifecycle Event
+
+## Advance
+
+### ConfiguresHook Event
+
+> It is recommended that you only implement this for `ConfigurableHook` hooks.
+
+In the event you want to preform advance logic to configure the hook as is done with the available Hooks you can create a new Event and Listener in order to configure your custom Hook. In order to do so ensure you override the `for` method in your hook as follows.
+
+```php
+// Hook
+public function for($event)
+{
+    return parent::for($event);
+    event(\App\Events\ConfiguresCustomHookEvent($this->event, $this));
+    return $this;
```

Comment:
This code is simply not right, You're returning and doing stuff after that.

Reply:
maybe parent::for just needs to be called and not returned?

Reply:
Good catch! Can't believe I missed that